src/common/abstractPromote.ts

```diff
@@ -48,20 +59,66 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
 
   protected async executePromotion(): Promise<PromotePipelineResult> {
     validateTestFlags(this.flags['test-level'], this.flags.tests);
-    const doceOrg: Org = await Org.create({ aliasOrUsername: this.flags['devops-center-username']?.getUsername() });
-    const pipelineStage: PipelineStage = await fetchAndValidatePipelineStage(
+    const doceOrg: Org = this.flags['devops-center-username'] as Org;
+    this.targetStage = await fetchAndValidatePipelineStage(
       doceOrg,
       this.flags['devops-center-project-name'],
       this.flags['branch-name']
     );
-    this.computeTargetStageId(pipelineStage);
+    this.sourceStageId = this.getSourceStageId();
+    const asyncOperationId: string = await this.requestPromotion(doceOrg);
+
+    if (this.flags.async) {
+      await DeployPipelineCache.set(asyncOperationId, {});
+      // TODO display async message
+    }
+
+    // TODO: move this to logger service
+    this.log(`Job ID: ${asyncOperationId}`);
+
+    return { jobId: asyncOperationId };
+  }
+
+  protected getTargetStage(): PipelineStage {
+    return this.targetStage;
+  }
+
+  private async requestPromotion(targetOrg: Org): Promise<string> {
+    this.buildPromoteOptions();
+    const req: HttpRequest = {
+      method: 'POST',
+      url: `${REST_PROMOTE_BASE_URL as string}${
+        this.targetStage.sf_devops__Pipeline__r.sf_devops__Project__c
+      }/pipelineName/${this.sourceStageId}`,
+      body: JSON.stringify({
+        changeBundleName: this.flags['bundle-version-name'],
```

Comment:
What happens if it is not a bundling stage? Is this undefined?

Reply:
Correct. The type of this.flags['bundle-version-name'] is string | undefined.

Reply:
Ok! I thought so